### PR TITLE
remove tuple_parser

### PR DESCRIPTION
**Pull request by nairbv**
Summary:

I could only find one usage of TupleParser and it doesn't seem worth maintaining just for that one usage.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30659 remove tuple_parser**

Differential Revision: [D18795979](https://our.internmc.facebook.com/intern/diff/D18795979)

### CODE REVIEW

File: torch/csrc/autograd/functions/init.cpp [hunk 1]
```diff
@@ -10,20 +10,21 @@
...
 struct DelayedErrorCtor {
   DelayedError* operator()(PyObject* args) {
-    std::string msg;
-    int num_inputs;
-
-    TupleParser parser(args, 2);
-    parser.parse(msg, "msg");
-    parser.parse(num_inputs, "num_inputs");
 
+    TORCH_CHECK(PyTuple_GET_SIZE(args) >= 2, "Too few arguments");
```
Comment (bddppq) [inline torch/csrc/autograd/functions/init.cpp]: I think here should check `== 2`?

Comment (nairbv) [reply torch/csrc/autograd/functions/init.cpp]: thanks!

File: torch/csrc/autograd/functions/init.cpp [hunk 2]
```diff
@@ -10,20 +10,21 @@
...
 struct DelayedErrorCtor {
   DelayedError* operator()(PyObject* args) {
-    std::string msg;
-    int num_inputs;
-
-    TupleParser parser(args, 2);
-    parser.parse(msg, "msg");
-    parser.parse(num_inputs, "num_inputs");
 
+    TORCH_CHECK(PyTuple_GET_SIZE(args) == 2, "Requires two arguments");
```
Comment (bddppq) [inline torch/csrc/autograd/functions/init.cpp]: nit: also put the actual size in the error message?

**Review by bddppq — APPROVED** [review]
LGTM

**Pull request closed without merging**